### PR TITLE
Adding version check to HUD

### PR DESCRIPTION
**Pull request by davidkpiano**
For #85 
screen shot 2017-05-23 at 3 16 24 pm

### CODE REVIEW

**Comment by davidkpiano** [comment]
Slight adjustments:
screen shot 2017-05-23 at 4 01 50 pm
screen shot 2017-05-23 at 4 01 43 pm

File: src/index.js [hunk 1]
```diff
@@ -11,15 +11,40 @@ require('./index.scss');
...
+            </g>
+        </g>
+    </svg>
+    `;
```
Comment (avanderhoorn) [inline src/index.js]: Nit, the arrow svg looks like its "tabbed" in 4 more spaces than this one, can we normalize?

File: src/index.js [hunk 2]
```diff
@@ -11,15 +11,40 @@ require('./index.scss');
...
+    let newVersion;
+    try {
+        const updateInfo = JSON.parse(localStorage.getItem('glimpseLatestVersion'));
+        if (updateInfo.latestVersion && updateInfo.latestVersion !== updateInfo.atTimeOfCheckVersion) {
+            newVersion = updateInfo.latestVersion;
+        }
+    } catch (e) {
+        // gulp
+    }
```
Comment (avanderhoorn) [inline src/index.js]: Is it worth breaking this out into ta function. Might make it a tad more readable.

Comment (davidkpiano) [reply src/index.js]: Fixed!

File: src/index.js (hunk 1 above)
Comment (davidkpiano) [reply src/index.js]: Fixed!

**Comment by legomushroom** [comment]
The ⚠️ sign supposed to change color regarding current version, is this working?

File: src/index.js [hunk 3]
```diff
@@ -10,16 +10,47 @@ require('./index.scss');
...
-    var arrow = `
+    const url = util.resolveClientUrl(util.currentRequestId(), true);
+    const arrow = `
         <svg class="glimpse-arrow" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 2048 2048">
             <path class="glimpse-arrow-path" d="M1022,2H2046V1026H1918V221L91,2047,1,1957,1827,130H1022V2Z"/>
         </svg>
     `;
+    const icon = `
+        <svg viewBox="0 0 51 42" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+            <g stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
```
Comment (legomushroom) [inline src/index.js]: Did you try to optimize the SVG source?

**Comment by legomushroom** [comment]
Few questions, LGTM.

File: src/index.js (hunk 3 above)
Comment (davidkpiano) [reply src/index.js]: No

I am ashamed

**Comment by avanderhoorn** [comment]
> The ⚠️ sign supposed to change color regarding current version, is this working?

Do you mean the part of the feature that try to get the users attention as more time goes on?

**Comment by avanderhoorn** [comment]
If so we don't need that right now.

**Comment by davidkpiano** [comment]
> The ⚠️ sign supposed to change color regarding current version, is this working?

This is just part of the SVG. No colors change. And this icon only shows when there's a new version available. There's no need to show the icon when everything's good (on current version).

**Comment by legomushroom** [comment]
I thought that the idea was that that icon is `greenish-yellow` at the beginning and it incrementally changes towards `red` for each new version available, just like Chrome does with the update notice. Of course, it should not show up at all if the Glimpse is up to date.

**Comment by avanderhoorn** [comment]
It was but for the moment this will do. Think of it as v1 of version notification, that will be v2.

**Comment by nikmd23** [comment]
This looks great. I'd request the icon be yellow (while it's color is static). Red should be reserved for errors or dire situations.

**Comment by avanderhoorn** [comment]
@davidkpiano when you get a moment, send in as a separate PR.

**Pull request merged**